src/nvim/eval.c

```diff
@@ -15577,6 +15577,53 @@ static void f_split(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   p_cpo = save_cpo;
 }
 
+/// "stdpath()" helper for list results
+void stdpath_list_helper(const XDGVarType xdg, typval_T *rettv)
+{
+  const void *iter = NULL;
+  list_T *const list = tv_list_alloc();
+  rettv->v_type = VAR_LIST;
+  rettv->vval.v_list = list;
+  list->lv_refcount++;
+  char *const dirs = stdpaths_get_xdg_var(xdg);
+  do {
+    size_t dir_len;
+    const char *dir;
+    iter = vim_colon_env_iter(dirs, iter, &dir, &dir_len);
```

Comment:
You need to merge in master because `vim_colon_env_iter` is no longer there (now `vim_env_iter` and colon may be the first argument).